tool/tsh/resolve_default_addr.go

```diff
@@ -47,10 +47,6 @@ const maxPingBodySize = 16 * 1024
 // level. Note that it is still the caller's responsibility to close the body
 // stream.
 func logResponseBody(level logrus.Level, bodyStream io.Reader) {
-	if log.Logger.Level < level {
```

Comment:
I think I would rather read the body outside of this function, removing logResponseBody in favor of a log.Debug statement. This is a strange side-effect to have in a logging function.

Draining the body is a good practice anyway, so we should do it in even when we are not logging errors. Something like `io.Copy(io.Discard, rsp.Body)` should do the trick.

Reply:
I agree with the suggestion, but I don't think that in this case `ioCopy()` is the best solution. We basically want to read the request body in case of error, so we can log it and discard it in case of success.
First of all I think we should leave the `io.LimitReader()` as we don't want to keep reading a response that may never end. Second, to make the logic simpler I think that always reading the body make the code simpler in this case.